addons/hr_timesheet/models/hr_timesheet.py

```diff
@@ -68,6 +68,28 @@ def _domain_employee_id(self):
     partner_id = fields.Many2one(compute='_compute_partner_id', store=True, readonly=False)
     readonly_timesheet = fields.Boolean(compute="_compute_readonly_timesheet", compute_sudo=True, export_string_translation=False)
     milestone_id = fields.Many2one('project.milestone', related='task_id.milestone_id')
+    message_partner_ids = fields.Many2many('res.partner', compute='_compute_message_partner_ids',
+                                           search='_search_message_partner_ids', readonly=True)
+
+    def _search_message_partner_ids(self, operator, value):
+        followed_ids_by_model = dict(self.env['mail.followers']._read_group([
+            ('partner_id', operator, value),
+            ('res_model', 'in', ('project.project', 'project.task')),
+        ], ['res_model'], ['res_id:array_agg']))
+        if not followed_ids_by_model:
+            return expression.FALSE_DOMAIN
+        domains = []
+        if project_ids := followed_ids_by_model.get('project.project'):
+            domains.append([('project_id', 'in', project_ids)])
+        if task_ids := followed_ids_by_model.get('project.task'):
+            domains.append([('task_id', 'in', task_ids)])
+        domain = expression.OR(domains)
+        return domain
+
+    @api.depends('project_id.message_partner_ids', 'task_id.message_partner_ids')
+    def _compute_message_partner_ids(self):
+        for line in self:
+            line.message_partner_ids = line.task_id.message_partner_ids.ids + line.project_id.message_partner_ids.ids
```

Comment:
Equivalent to: 
```suggestion
            line.message_partner_ids = line.task_id.message_partner_ids + line.project_id.message_partner_ids
```

Also, no need to do an union in the case the follower is in the project and task? 🧐